lib/ui/compositing/scene.cc

```diff
@@ -102,7 +98,41 @@ Dart_Handle Scene::toImage(uint32_t width,
   return Picture::RasterizeToImage(picture, width, height, raw_image_callback);
 }
 
-std::unique_ptr<flutter::LayerTree> Scene::takeLayerTree() {
+void Scene::RasterizeToImageSync(uint32_t width,
+                                 uint32_t height,
+                                 Dart_Handle raw_image_handle) {
+  auto* dart_state = UIDartState::Current();
+  auto unref_queue = dart_state->GetSkiaUnrefQueue();
+  auto snapshot_delegate = dart_state->GetSnapshotDelegate();
+  auto raster_task_runner = dart_state->GetTaskRunners().GetRasterTaskRunner();
+
+  auto image = CanvasImage::Create();
+  auto dl_image = DlDeferredImageGPU::Make(SkISize::Make(width, height));
+  image->set_image(dl_image);
+
+  fml::TaskRunner::RunNowOrPostTask(
+      raster_task_runner,
+      [snapshot_delegate, unref_queue, dl_image = std::move(dl_image),
+       layer_tree = layer_tree_, width, height]() {
+        auto display_list =
+            layer_tree->Flatten(SkRect::MakeWH(width, height),
+                                snapshot_delegate.get()->GetTextureRegistry());
```

Comment:
need a to check snapshot_delegate. It might have been collected by now.

Reply:
Doneish. Do I need to set anything on the dl_image, or could we assume we're shutting down now?